src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Combinators.hs

```diff
@@ -119,3 +127,12 @@ runInvert r = do
     where
       isZero :: forall a . (Ring a, Eq (Bool a) a, Conditional (Bool a) a) => a -> a
       isZero x = bool @(Bool a) zero one (x == zero)
+
+embedVarIndex :: Arithmetic a => Natural -> ArithmeticCircuit 1 a
+embedVarIndex n = ArithmeticCircuit { acCircuit = mempty { acInput = [ n ]}, acOutput = pure n}
+
+embedVarIndexV :: (Arithmetic a, KnownNat n) => Natural -> ArithmeticCircuit n a
+embedVarIndexV n = ArithmeticCircuit { acCircuit = mempty { acInput = [ n ]}, acOutput = pure n}
+
+getAllVars :: MultiplicativeMonoid a => Circuit a -> [Natural]
```

Comment:
Why is `MultiplicativeMonoid a` constraint needed?

Reply:
This is necessary for the function `variables`